antiope-shell/src/Antiope/Shell/S3.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections       #-}
+
+module Antiope.Shell.S3
+  ( putFile
+  ) where
+
+import Antiope.S3.Types      (S3Uri)
+import Control.Monad.Except
+import Data.Monoid           ((<>))
+import Data.Text             as T (Text, pack, unpack)
+import Network.AWS.Data.Text (toText)
+
+import qualified System.Exit    as IO
+import qualified System.Process as IO
+
+-- | Puts file into a specified S3 bucket
+putFile :: MonadIO m
+  => S3Uri        -- ^ File name on S3
+  -> FilePath         -- ^ Source file path
+  -> ExceptT Text m ()   -- ^ Etag when the operation is successful
+putFile s3Uri filePath = do
+  exitCode <- liftIO $ IO.rawSystem "aws" ["s3", "cp", filePath, T.unpack (toText s3Uri)]
```

Comment:
Does this spam output to stdout?

Reply:
Probably does.  Should it not?

Reply:
By default aws s3 cp will print lots of ansi control codes for a pretty output. This will make any jobs using it have garbage in their logs. What does it look like when you run it?

Reply:
This is what it looks like:

```
λ> runExceptT $ Antiope.Shell.S3.putFile uri "antiope-shell.cabal"
upload: ./antiope-shell.cabal to s3://jky-mayhem/blah
Right ()
```

Reply:
This could be what you're concerned about:

```
λ> runExceptT $ Antiope.Shell.S3.putFile uri "/Users/jky/wrk/haskell-works/hw-uri/7g.bin"
Completed 14.8 MiB/7.0 GiB (1.6 MiB/s) with 1 file(s) remaining
```

Reply:
I added a `--quiet` flag to ensure this doesn't happen.